build.gradle

```diff
@@ -95,10 +107,6 @@ project('common') {
         testCompile project(':test:testcommon')
         compile group: 'commons-io', name: 'commons-io', version: commonsioVersion
         compile group: 'com.google.guava', name: 'guava', version: guavaVersion
-        compile group: 'com.github.spotbugs', name: 'spotbugs-annotations', version: spotbugsAnnotationsVersion
-        compile group: 'net.jcip', name: 'jcip-annotations', version: jcipAnnotationsVersion
-
-        compile group: 'org.projectlombok', name: 'lombok', version: lombokVersion
```

Comment:
I understand that we are implicitly changing this to `compileOnly` by applying the plugin. What's the advantage of using the plugin vs. just changing across the gradle projects? Is it that we guarantee uniformity automatically? I'm not sure I see a big advantage in doing it for a single dependency, but maybe I'm missing something.

Reply:
I wanted to use the lombok plugin because it is easier than the alternative https://projectlombok.org/setup/gradle which requires to add apt processing explicitly to all of our targets. 
At first I tried just changing the annotation packages to be compile only. This did not work. Lombok depends on the spotbugs annotations so that it's generated code can use them. Before we had a lot of targets not declaring a dependency on the annotations but depending on lombok.

So when I changed the downstream packages to have a compile only dependency, suddenly it was missing from their classpath and the lombok errored with this cryptic error message:
https://github.com/rzwitserloot/lombok/issues/1079

This was extremely confusing, took a while to track down, and would be super easy to trigger again accidentally. For this reason I decided it would be better if it were all in one place next to the lombok declaration so that it would be applied uniformly across all the packages.